lib/input/input_controller.h

```diff
@@ -68,6 +77,8 @@ struct InputController {
 
     static double mouse_x, mouse_y;
     static double mouse_delta_x, mouse_delta_y;
+
+    static GLFWwindow *active_window_;
```

Comment:
Storing active window inside input controller seems wrong to me... I think there are better solutions available.

May be we should leave `InputController` as it is, and create a static class `WindowManager` instead?

```C++
struct WindowManager {
    static GLFWwindow* get_active_window();
    static void set_active_windw(WindowId identifier);

    static WindowId construct_game_window(/* Purpose, settings, styles? */);

    // Other window management utilities?
    // . . .
};
```

This way `InputController` does not take a role of `WindowManager` of the project, and still can be accessed from anywhere with explicit specification of which window's even processing behavior is getting changed.

```C++
// Somewhere deep inside the game's code

InputManager::set_cursor_mode(WindowManager::get_active_window(), CursorMode::Camera);
```